project/src/pages/Login/index.tsx

```diff
@@ -1,34 +1,74 @@
+import { LoginData } from '../../types/login';
 
 
-export const Login = () => (
+export const Login = () => {
+  const dispatch = useAppDispatch();
 
-  <div className="user-page">
-    <header className="page-header user-page__head">
-      <Logo />
+  const loginRef = useRef<HTMLInputElement | null>(null);
+  const passwordRef = useRef<HTMLInputElement | null>(null);
 
-      <h1 className="page-title user-page__title">Sign in</h1>
-    </header>
+  const onSubmit = (loginData: LoginData) => {
+    dispatch(fetchLoginAction(loginData));
+  };
 
-    <div className="sign-in user-page__content">
-      <form action="#" className="sign-in__form">
-        <div className="sign-in__fields">
-          <div className="sign-in__field">
-            <input className="sign-in__input" type="email" placeholder="Email address" name="user-email" id="user-email" />
-            <label className="sign-in__label visually-hidden" htmlFor="user-email">Email address</label>
+  const submitLoginFormHandle = (evt: React.FormEvent<HTMLFormElement>) => {
+    evt.preventDefault();
+
+    if (loginRef.current !== null && passwordRef.current !== null) {
+      onSubmit({
+        email: loginRef.current.value,
+        password: passwordRef.current.value,
+      });
+    }
+  };
+
+
+  return (
+    <div className="user-page">
+      <header className="page-header user-page__head">
+        <Logo />
+
+        <h1 className="page-title user-page__title">Sign in</h1>
+      </header>
+
+      <div className="sign-in user-page__content">
+        <form onSubmit={submitLoginFormHandle} action="#" className="sign-in__form">
+          <div className="sign-in__fields">
+            <div className="sign-in__field">
+              <input
+                ref={loginRef}
+                className="sign-in__input"
+                type="email"
+                placeholder="Email address"
+                name="user-email"
+                id="user-email"
+              />
+              <label className="sign-in__label visually-hidden" htmlFor="user-email">Email address</label>
+            </div>
+            <div className="sign-in__field">
+              <input
+                ref= {passwordRef}
+                className="sign-in__input"
+                type="password"
+                placeholder="Password"
+                name="user-password"
+                id="user-password"
+              />
+              <label className="sign-in__label visually-hidden" htmlFor="user-password">Password</label>
+            </div>
           </div>
-          <div className="sign-in__field">
-            <input className="sign-in__input" type="password" placeholder="Password" name="user-password" id="user-password" />
-            <label className="sign-in__label visually-hidden" htmlFor="user-password">Password</label>
+          <div className="sign-in__submit">
+            <button className="sign-in__btn" type="submit">Sign in</button>
```

Comment:
let's disable button while fetchLoginAction in progress

Reply:
fixed

Reply:
fixed

Reply:
fixed